internal/resource.go

```diff
@@ -63,13 +63,32 @@ func getUrl(u string, fileName string, ctx context.Context) (string, error) {
 	return fileName, nil
 }
 
-// GetUrlToDir downloads the given resource to the given directory and returns the path to it.
-func GetUrlToDir(u string, targetDir string, ctx context.Context) (string, error) {
+// GetUrlToDir downloads the given resource to a temporary file and returns the path to it.
+// Modify the GetUrlToDir function to ensure proper file cleanup
+func GetUrlToDir(u string, targetDir string, ctx context.Context) 
+(string, error) {
 	// create temporary name in the target directory.
 	h := sha256.New()
 	h.Write([]byte(u))
 	fileName := filepath.Join(targetDir, fmt.Sprintf(".%s", hex.EncodeToString(h.Sum(nil))))
-	return getUrl(u, fileName, ctx)
+
+	// Remove existing file if present
+	os.Remove(fileName)
+
+	// Create new file with immediate close
+	file, err := os.Create(fileName)
+	if err != nil {
+		return "", fmt.Errorf("failed to create file: %v", err)
+	}
+	file.Close()
+
+	fileName, err = getUrl(u, fileName, ctx)
+	if err != nil {
+		os.Remove(fileName)
```

Comment:
I agree with the intent here but I don't think this cleanup belongs here. If we really want to do it we should do it in `getUrl`: in the error case there, check if the file exists and if it does, remove it.

Reply:
I'm still not sure about this. Why do we need this code at all?

Reply:
I don't see the result of this discussion in the code.